src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -710,4 +711,13 @@ protected int doHashCode() {
     public String getWriteableName() {
         return NAME;
     }
+
+    @Override
+    protected QueryBuilder doRewrite(QueryRewriteContext queryShardContext) throws IOException {
+        // rewrite filter query if it exists to avoid runtime errors in next steps of query phase
+        if (Objects.nonNull(filter)) {
+            filter = filter.rewrite(queryShardContext);
```

Comment:
can it happen that after filter.rewrite is done the filter query is null?

Reply:
yes, it's possible for a simple filters with invalid clauses (like initial reported case with non existing field)